src/main/java/org/terasology/gooeyDefence/TowerManager.java

```diff
@@ -87,6 +87,61 @@ private static long getTargeterId(String eventId) {
         return Long.parseLong(id);
     }
 
+    /**
+     * Get the drain caused by all the targeters on a tower
+     *
+     * @param towerComponent The TowerComponent of the tower entity
+     * @return The total drain. Zero if the tower has no targeters
+     */
+    public static int getTargeterDrain(TowerComponent towerComponent) {
+        int drain = 0;
+        for (EntityRef emitterEntity : towerComponent.targeter) {
+            TowerTargeter emitter = DefenceField.getComponentExtending(emitterEntity, TowerTargeter.class);
+            drain += emitter.getDrain();
+        }
+        return drain;
+    }
+
+    /**
+     * Get the drain caused by all the effector on a tower
+     *
+     * @param towerComponent The TowerComponent of the tower entity
+     * @return The total drain. Zero if the tower has no effector
+     */
+    public static int getEffectorDrain(TowerComponent towerComponent) {
+        int drain = 0;
+        for (EntityRef effectorEntity : towerComponent.effector) {
```

Comment:
would plural be better for the field name in towerComponent?

The three accumulator-methods can probably be written as a stream.map(.map).sum operation, however this is more a style than a worse/better question ;) The way it is is also okay.

Reply:
Now that you've pointed out the lack of a plural I can't unsee it. I'll fix it directly on the main branch after this is merged however, because it'll touch a lot of files.

I've also changed the methods to use map. After looking at it it looks cleaner to do it that way.